src/applications/proxy-rewrite/header-footer-loader/header-footer-loader-entry.jsx

```diff
@@ -0,0 +1,165 @@
+import cookie from 'cookie';
+import _ from 'lodash';
+
+import redirectIfNecessary from '../redirects';
+import { proxyRewriteWhitelist } from './proxy-rewrite-whitelist.json';
+import BUCKETS from '../../../site/constants/buckets';
+
+function createMutationObserverCallback() {
+  // Find native header, footer, etc based on page path
+  const DEPRECATED_SELECTOR_CONFIG = [
+    {
+      path: /.*/,
+      selector:
+        'header.row.main-header-wrap, div#top-nav-wrapper, div#main-header, div#footer-effect',
+    },
+  ];
+
+  let deprecatedSelector;
+  for (const config of DEPRECATED_SELECTOR_CONFIG) {
+    if (document.location.pathname.match(config.path) !== null) {
+      deprecatedSelector = config.selector;
+      break;
+    }
+  }
+
+  // Hide native elements when they're added to the DOM
+  return function mutationObserved(mutations) {
+    mutations.forEach(mutation => {
+      Array.from(mutation.addedNodes).forEach(node => {
+        if (node.nodeType === Node.ELEMENT_NODE) {
+          if (node.tagName === 'BODY') {
+            node.classList.add('merger');
+          } else if (node.matches(deprecatedSelector)) {
+            node.classList.add('brand-consolidation-deprecated');
+          }
+        }
+      });
+    });
+  };
+}
+
+function getProxyRewriteCookieValue(
+  cookies = document.cookie,
+  parseCookie = cookie.parse,
+) {
+  return parseCookie(cookies).proxyRewrite;
+}
+
+function getMatchedWhitelistItem(whitelist = proxyRewriteWhitelist) {
+  const { hostname, pathname } = window.location;
+
+  return whitelist.find(
+    whitelistItem =>
+      whitelistItem.hostname === hostname &&
+      pathname.startsWith(whitelistItem.pathnameBeginning),
+  );
+}
+
+function shouldActivateInjectedAssets(whitelistItem, proxyRewriteCookieValue) {
+  if (whitelistItem === undefined) {
+    return false;
+  }
+
+  if (whitelistItem.cookieOnly) {
+    return proxyRewriteCookieValue;
+  }
+
+  return true;
+}
+
+function startHeaderMutationObserver() {
+  const observer = new MutationObserver(createMutationObserverCallback());
+  observer.observe(document, {
+    attributes: true,
+    childList: true,
+    subtree: true,
+  });
+
+  document.addEventListener('DOMContentLoaded', _e => {
+    observer.disconnect();
+  });
+}
+
+function getHostname() {
+  // default to vagovprod
+  const environment =
+    new URLSearchParams(window.location.search).get('hfbuild') || 'vagovprod';
```

Comment:
`URLSearchParams` [isn't on IE11](https://caniuse.com/#feat=urlsearchparams) so you'd need to use a polyfill for it. Alternately you could just use a regex to pull out the param, something like `window.location.search.match(/\bhfbuild=(\w+)/)` and it will either return null or an array with the buildType in [1].

As a one-liner it could look like:
```js
const environment =   
    (window.location.search.match(/\bhfbuild=(\w+)/) || [])[1] || 'vagovprod'
```

Reply:
Thanks I'll use that